pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -82,3 +82,44 @@ func resourceName(name string) string {
 	parts := strings.Split(name, "/")
 	return parts[len(parts)-1]
 }
+
+func (ca *cloudCIDRAllocator) NetworkToNodes(network *networkv1.Network) error {
+	k8sNodesList, err := ca.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error fetching nodes: %v", err)
+	}
+	// Need to reschedule all Nodes, since we do not know which one holds the
+	// new Network.
+	for _, node := range k8sNodesList {
+		if network != nil {
+			// filter out nodes that are not part of network
+			if node.Annotations == nil {
+				// skip node w/o any annotation, not possible for it to have any MN
+				continue
+			}
+			northIntfAnn, ok := node.Annotations[networkv1.NorthInterfacesAnnotationKey]
+			if !ok {
+				// skip node w/o "north-interfaces" annotation, no MN
+				continue
+			}
+			northIntf, err := networkv1.ParseNorthInterfacesAnnotation(northIntfAnn)
+			// if err!=nil means there is some format issue with the annotation, lets
+			// re-generate it for that node
+			if err == nil {
+				found := false
+				for _, ele := range northIntf {
+					if ele.Network == network.Name {
+						found = true
+						break
+					}
+				}
+				if !found {
+					// node is not part of this network
+					continue
+				}
+			}
+		}
+		_ = ca.AllocateOrOccupyCIDR(node)
```

Comment:
I think we should be careful with this function here. It wasn't designed to be called over and over again especially for podCIDR allocation. I went through the worker and it looks like it will work but I am worried because I don't know the implications too well.

Let's say this is called for a node which is unrelated to the network being added. Here is what will happen:
1. We will fetch the node instance from GCP
2. We will compare the podCIDR on the node object to the podCIDR from the node instance. If these match (which they should) then we will update the NetworkUnavailable status on the node. 
3. If this is a node with an additional interface, then the northInterface and networks annotation along with the  capacity status are updated again.

One option is that we use a separate queue and channel for updating networks only.

Reply:
we need to rework the reconcile function then, so that it could be rerun cheaply at any point. Having multiple functions for slightly different cases leads to inconsistencies.